File: CS/Lecture.cs
namespace StudentManagementSystem;

using System.Collections.Generic;

public class Lecture
{
    private string _name;
    private string _address;
    private string _timeStart;
    private string _timeStop;
    private List<Student> _students;
    private Teacher _homeroomTeacher;
    private StudentPresident? _studentPresident;
    private StudentVicePresident? _studentVicePresident;
    private Dictionary<String, byte> _grades;
    private Dictionary<String, Dictionary<String, bool>> _attendanceList;

    public Lecture(string name, string address, string timeStart, string timeStop, Teacher teacher)
    {
        _name = name;
        _address = address;
        _timeStart = timeStart;
        _timeStop = timeStop;
        _students = new();
        _homeroomTeacher = teacher;
				_grades = new();
				_attendanceList = new();
    }

    public string Name => _name;
    public string Address => _address;
    public string Start => _timeStart;
    public string Stop => _timeStop;
    public List<Student> Students => _students;
    public Teacher HomeroomTeacher { get => _homeroomTeacher; set => _homeroomTeacher = value; }
    public StudentPresident? StudentPresident { get => _studentPresident; set => _studentPresident = value; }
    public StudentVicePresident? StudentVicePresident { get => _studentVicePresident; set => _studentVicePresident = value; }
    public Dictionary<String, byte> Grades => _grades;
    public Dictionary<String, Dictionary<String, bool>> AttendanceList => _attendanceList;
}

File: Rust/src/lecture.rs
use crate::student::{Student, StudentPresident, StudentVicePresident};
use crate::teacher::Teacher;
use std::collections::HashMap;

pub type Grades = HashMap<String, i8>;
pub type AttendanceChecker = HashMap<String, bool>;
pub type AttendanceList = HashMap<String, AttendanceChecker>;

pub struct Lecture {
	name: String,
	address: String,
	time_start: String,
	time_stop: String,
	homeroom_teacher: Teacher,
	students: Vec<Student>,
	student_president: Option<StudentPresident>,
	student_vicepresident: Option<StudentVicePresident>,
	grades: Grades,
	attendance_list: AttendanceList
}

impl Lecture {
	pub fn new(name: &str, address: &str, time_start: &str, time_stop: &str, teacher: &Teacher) -> Self {
		Self {
			name: name.into(),
			address: address.into(),
			time_start: time_start.into(),
			time_stop: time_stop.into(),
			homeroom_teacher: teacher.clone(),
			student_president: None,
			student_vicepresident: None,
			students: Vec::new(),
			grades: Grades::new(),
			attendance_list: AttendanceList::new(),
		}
	}

	pub fn get_name(&self) -> &str {
		&self.name
	}

	pub fn get_address(&self) -> &str {
		&self.address
	}

	pub fn get_time_start(&self) -> &str {
		&self.time_start
	}

	pub fn get_time_stop(&self) -> &str {
		&self.time_stop
	}

	pub fn get_students(&self) -> &Vec<Student> {
		&self.students
	}

	pub fn get_student_vicepresident(&self) -> &Option<StudentVicePresident> {
		&self.student_vicepresident
	}

	pub fn get_student_president(&self) -> &Option<StudentPresident> {
		&self.student_president
	}

	pub fn get_grades(&mut self) -> &mut HashMap<String, i8> {
		&mut self.grades
	}

	pub fn get_attendance_list(&mut self) -> &mut HashMap<String, HashMap<String, bool>> {
		&mut self.attendance_list
	}
}
